k9mail-library/src/main/java/com/fsck/k9/mail/store/imap/ImapFolder.java

```diff
@@ -1273,6 +1273,28 @@ public void expunge() throws MessagingException {
         }
     }
 
+    @Override
+    public void expungeMessages(List<? extends Message> messages) throws MessagingException {
+        if (messages.isEmpty()) {
+            return;
+        }
+
+        checkOpen(); //only need READ access
+
+        String[] uids = new String[messages.size()];
```

Comment:
this bock occurs in other places in the code, so it's probably worth its own method